common/tqid/task_queue_id.go

```diff
@@ -357,6 +360,81 @@ func (p *NormalPartition) RoutingKey() string {
 	return fmt.Sprintf("%s:%s:%d", p.NamespaceId(), p.RpcName(), p.TaskType())
 }
 
+// ValidateTaskQueue validates a TaskQueue object.
+// It checks the TaskQueue's name for emptiness, length, UTF-8 validity, and whitespace.
+// For sticky queues, it also validates the NormalName.
+// If the name is empty and defaultVal is provided, it sets the name to defaultVal.
+// If the Kind is unspecified, it sets it to NORMAL.
+//
+// Parameters:
+//   - taskQueue: The TaskQueue to validate. If nil, returns an error.
+//   - defaultVal: Default name to use if taskQueue name is empty.
+//   - maxIDLengthLimit: Maximum allowed length for the TaskQueue name.
+//
+// Returns an error if validation fails, nil otherwise.
+func ValidateTaskQueue(
+	taskQueue *taskqueuepb.TaskQueue,
+	defaultVal string,
+	maxIDLengthLimit int,
+) error {
+	if taskQueue == nil {
+		return serviceerror.NewInvalidArgument("TaskQueue is not set.")
+	}
+
+	enums.SetDefaultTaskQueueKind(&taskQueue.Kind)
```

Comment:
I think it may be slightly confusing that a validation method is modifying the object it is validating, even if it is just setting defaults. Not a big deal, but I would keep the semantics of the previous methods and return the validated task queue to make it a bit clearer.

Reply:
I think the docs make it clear:

```
// If the name is empty and defaultVal is provided, it sets the name to defaultVal.
// If the Kind is unspecified, it sets it to NORMAL.
```

That being said, I am new to Go. If this is a confusing contract, I can change it as you describe

Reply:
I would rename this function to `SetDefaultTaskQueueParams`.

Reply:
At its heart, the function is doing two things:
1. Validating object state
2. Normalizing state, where pragmatic

Maybe this is the Java guy in me, but `NormalizeAndValidate` would clearly convey the function intent. The `TaskQueue` method parameter makes it clear _what_ is being normalized and validated. Thoughts?

Reply:
I like this.